osquery/config/tests/packs_tests.cpp

```diff
@@ -0,0 +1,138 @@
+  std::stringstream json;
+  json << content;
+  pt::ptree tree;
+  pt::read_json(json, tree);
+  return tree;
+}
+
+pt::ptree getKernelPack() {
+  auto tree = getExamplePacksConfig();
+  auto packs = tree.get_child("packs");
+  std::stringstream json_stream;
+  pt::write_json(json_stream, packs);
+  auto json = json_stream.str();
+  json_stream.str(std::string());
+
+  return packs.get_child("kernel");
+}
+
+pt::ptree getFoobarPack() {
+  auto tree = getExamplePacksConfig();
+  auto packs = tree.get_child("packs");
+  std::stringstream json_stream;
+  pt::write_json(json_stream, packs);
+  auto json = json_stream.str();
+  json_stream.str(std::string());
+
+  return packs.get_child("foobar");
+}
+
+pt::ptree getFoobazPack() {
+  auto tree = getExamplePacksConfig();
+  auto packs = tree.get_child("packs");
+  std::stringstream json_stream;
+  pt::write_json(json_stream, packs);
+  auto json = json_stream.str();
+  json_stream.str(std::string());
+
+  return packs.get_child("foobaz");
+}
+
+TEST_F(PacksTests, test_parse) {
+  auto tree = getExamplePacksConfig();
+  EXPECT_EQ(tree.count("packs"), 1);
+}
+
+TEST_F(PacksTests, test_should_pack_execute) {
+  auto kpack = Pack("kernel", getKernelPack());
+  EXPECT_TRUE(kpack.shouldPackExecute());
+
+  auto fpack = Pack("foobar", getFoobarPack());
+  EXPECT_FALSE(fpack.shouldPackExecute());
+}
+
+TEST_F(PacksTests, test_get_discovery_queries) {
+  std::vector<std::string> expected;
+  auto kpack = Pack("kernel", getKernelPack());
+  expected = {};
+  EXPECT_EQ(kpack.getDiscoveryQueries(), expected);
+
+  expected = {"select pid from processes where name = 'foobar';"};
+  auto fpack = Pack("foobar", getFoobarPack());
+  EXPECT_EQ(fpack.getDiscoveryQueries(), expected);
+}
+
+TEST_F(PacksTests, test_platform) {
+  auto fpack = Pack("foobar", getFoobarPack());
+  EXPECT_EQ(fpack.getPlatform(), "darwin");
+}
+
+TEST_F(PacksTests, test_version) {
+  auto fpack = Pack("foobar", getFoobarPack());
+  EXPECT_EQ(fpack.getVersion(), "1.5.0");
+}
+
+TEST_F(PacksTests, test_name) {
+  auto fpack = Pack("foobar", getFoobarPack());
+  fpack.setName("foobar");
+  EXPECT_EQ(fpack.getName(), "foobar");
+}
+
+TEST_F(PacksTests, test_check_platform) {
+  auto fpack = Pack("foobar", getFoobarPack());
+  if (kSDKPlatform == "darwin") {
+    EXPECT_TRUE(fpack.checkPlatform());
+  } else {
+    EXPECT_FALSE(fpack.checkPlatform());
+  }
+}
+
+TEST_F(PacksTests, test_check_version) {
+  auto fpack = Pack("foobar", getFoobarPack());
+  EXPECT_TRUE(fpack.checkVersion());
+
+  auto zpack = Pack("foobaz", getFoobazPack());
+  EXPECT_FALSE(zpack.checkVersion());
+}
+
+TEST_F(PacksTests, test_schedule) {
+  auto fpack = Pack("foobar", getFoobarPack());
+  EXPECT_EQ(fpack.getSchedule().size(), 1);
+}
+
+TEST_F(PacksTests, test_discovery_cache) {
+  auto pack = Pack("foobar", getFoobarPack());
+  for (int i = 0; i < 5; i++) {
+    pack.checkDiscovery();
+  }
+  auto stats = pack.getStats();
+  EXPECT_EQ(stats.total, 5);
```

Comment:
In most of these tests the content of the config, pack, and other JSON is opaque to the test reader. Short explanations of why we expect each would be fun, definitely a nit.